src/App.js

```diff
@@ -1,62 +1,77 @@
-        setDescription("");
-        setUser("");
-    }
+        if (title.trim() === "" || description.trim() === "" || user.trim() === "") {
+            setModalActive(false);
+            return;
+        }
+
+        const newTask = {
+            id: nextId,
+            title: title,
+            description: description,
+            user: user
+        };
+
+        setNextId(nextId + 1);
+
+        const tasks = JSON.parse(localStorage.getItem("tasks")) || [];
+        localStorage.setItem("tasks", JSON.stringify([...tasks, newTask]));
+
+        setTaskList((prevTaskList) => [...prevTaskList, newTask]);
+
+        setModalActive(false);
+    };
+
+    const handleCreateTask = () => {
+        setReset();
+        setModalActive(true);
+    };
+
+    const handleCloseModal = () => {
+        setModalActive(false);
+    };
 
     return (
         <div className="app">
-            <header className="header">
-                <div className="header-title">TRELLO</div>
-                <button className="create-task-button" onClick={() => setModalActive(true)}>
-                    Create New Task
-                </button>
-            </header>
-            <Modal active={modalActive} setActive={setModalActive}>
-                <form className="form" onSubmit={handleSubmit}>
-                    <div className="form-title">
-                        <h1 className="title">Create task:</h1>
-                        <button className="close-button">&#10006;</button>
-                    </div>
-
-                    <input
-                        type="text"
-                        placeholder="Title"
-                        value={title}
-                        onChange={(e) => setTitle(e.target.value)}
-                    />
-                    <input
-                        type="text"
-                        placeholder="Description"
-                        value={description}
-                        onChange={(e) => setDescription(e.target.value)}
-                    />
-                    <input
-                        type="text"
-                        placeholder="User"
-                        value={user}
-                        onChange={(e) => setUser(e.target.value)}
-                    />
-                    <button className="button-create">Create</button>
-                </form>
+            <Header handleCreateTask={handleCreateTask} />
+            <Modal active={modalActive} setActive={setModalActive} handleCloseModal={handleCloseModal} title="Create task:">
+                <TaskForm
+                    title={title}
+                    description={description}
+                    user={user}
+                    onTitleChange={(e) => setTitle(e.target.value)}
+                    onDescriptionChange={(e) => setDescription(e.target.value)}
+                    onUserChange={(e) => setUser(e.target.value)}
+                    onSubmit={handleSubmit}
+                    isFormValid={isFormValid}
```

Comment:
do we really need to call useTaskForm here and then give to this component all this props? maybe its better to call it inside TaskForm?